lib/modules/filteReddit.js

```diff
@@ -787,30 +808,31 @@ function scanEntries(ele) {
 		isCurrentSubreddit('all') || currentDomain() || isCurrentMultireddit('me/f/all');
 	const onSavedPage = excludeSaved.test(location.href);
 
-	things.forEach(thing => {
+	await things::forEachChunked(async thing => {
 		let postSubreddit, currSub;
 		if (thing.isPost() && !onSavedPage) {
 			const postTitle = thing.getTitle();
 			const postDomain = thing.getPostDomain();
 			const postFlair = thing.getPostFlairText();
+			const postLink = thing.getPostLink().href;
 
 			let filtered = executeCustomFilters(thing);
 
 			currSub = currentSubreddit() ? currentSubreddit().toLowerCase() : null;
 			postSubreddit = thing.getSubreddit() || currSub;
 
 			if (!filtered) filtered = filterTitle(postTitle, postSubreddit);
+			if (!filtered) filtered = removeVisitedActive && await filterVisited(postLink);
```

Comment:
I am a bit concerned about having a potentially slow async operation in the middle of filtering, since right now filters can run before first paint, which I doubt could happen with `filterVisited` enabled.

Maybe do `filterVisited` in a second `things::forEachChunked` pass directly after this one, so the fast filters are always fast?